pkg/kubeconfig/create.go

```diff
@@ -56,7 +56,9 @@ func CreateKubeConfigFromServiceAccount(apiServerConfigProvider utils.ApiServerC
 // 1. from the ClusterConfig
 // 2. defaults to 6443.
 func GetApiServerURL(apiServerConfigProvider utils.ApiServerConfigProvider, clientset kubernetes.Interface) (string, error) {
-	address := apiServerConfigProvider.GetAPIServerConfig().Address
+	config := apiServerConfigProvider.GetAPIServerConfig()
+	var port string
+	address := utils.GetHost(config.Address)
```

Comment:
It's not part of the PR but maybe the following logic to extract the fallback address can be encapsulated in a dedicated method for sake of clarity.